Makefile.inc

```diff
@@ -21,35 +27,57 @@ PKG_CPPFLAGS := \
 	$(shell pkg-config spdlog --cflags $(silence)) \
 	$(shell pkg-config fmt --cflags $(silence))
 
-SAL_CPPFLAGS += $(PKG_CPPFLAGS) \
+ifeq ($(WITH_SAL_KAFKA),1)
+  SAL_CPPFLAGS += $(PKG_CPPFLAGS)  \
+	-I${SAL_WORK_DIR}/MTM1M3/cpp/src -I${SAL_WORK_DIR}/MTMount/cpp/src -I${SAL_WORK_DIR}/include \
+	-I${SAL_HOME}/include -I${LSST_SDK_INSTALL}/include -I${SAL_HOME}/include -I${LSST_SAL_PREFIX}/include -I${LSST_SAL_PREFIX}/include/avro
+else
+  SAL_CPPFLAGS += $(PKG_CPPFLAGS) \
 	-I${OSPL_HOME}/include -I${OSPL_HOME}/include/sys -I${OSPL_HOME}/include/dcps/C++/SACPP \
 	-I${SAL_WORK_DIR}/MTM1M3/cpp/src -I${SAL_WORK_DIR}/MTMount/cpp/src -I${SAL_WORK_DIR}/include \
 	-I${SAL_HOME}/include -I${LSST_SDK_INSTALL}/include
+endif
 
 PKG_LIBS := \
 	$(shell pkg-config yaml-cpp --libs $(silence)) -lstdc++fs
 
+
 CPP := g++ -std=c++17 -Wall -fPIE ${c_opts}
 
 # compile x86 simulator or real cRIO stuff
 ifdef SIMULATOR
   C := gcc -Wall ${c_opts}
-  CPP += -pedantic -DSIMULATOR
+  CPP += -DSIMULATOR
   PKG_LIBS += $(shell pkg-config spdlog --libs $(silence))
 else
   C := gcc -Wall -fmessage-length=0 ${c_opts}
   CPP += -fmessage-length=0
   PKG_LIBS += $(shell pkg-config spdlog --libs $(silence) | sed -E 's/-l([a-z0-9]*)/-l:lib\1.a/g')
 endif
 
-LIBS += $(PKG_LIBS) -ldl 
+ifeq ($(WITH_SAL_KAFKA),1)
+  LIBS += $(PKG_LIBS) -ldl -lpthread -L/usr/lib64/boost${BOOST_RELEASE} -lboost_filesystem -lboost_iostreams \
+	-lboost_program_options -lboost_system \
+	-L${LSST_SAL_PREFIX}/lib -lcurl -lrdkafka++ -lrdkafka -lavrocpp -lavro -ljansson
```

Comment:
-lboost etc. shall be placed after -lavro - -lavro requested some linking from boost, so the g++/linker will then properly know which boost functions are needed.